source/common/modules/markdown-editor/util/strip-duplicate-spaces.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        stripDuplicateSpaces function
+ * CVM-Role:        Utility function
+ * Maintainers:     Rich Douglas
+ * License:         GNU GPL v3
+ *
+ * Description:     This function strips duplicate spaces from a selected region
+ *                  of text.
+ *
+ * END HEADER
+ */
+
+import { type EditorView } from '@codemirror/view'
+import { type ChangeSpec } from '@codemirror/state'
+
+/**
+ * Utility to strip duplicate spaces from a selected region of text.
+ *
+ * @param {EditorView} view - The editor view
+ */
+export function stripDuplicateSpaces (view: EditorView): void {
+  const changes = view.state.selection.ranges
+    .reduce((allChanges, { from, to }) => {
+      const text = view.state.sliceDoc(from, to)
+
+      const strippedText = text.replaceAll(/ {2,}/g, ' ')
```

Comment:
I think I've done exactly what the issue description asked for: strip duplicate _spaces_.

Duplicate tabs and other "space-like" characters are not stripped.